### DISCUSSION TITLE
Splitting up 2FA specs

### DISCUSSION THREAD

**Discussion by janko** [jeremyevans/rodauth#459] [General]
When I'm making changes to 2FA or trying to understand how something works, I find it a bit difficult to navigate `spec/two_factor_spec.rb` due to its size. 2FA methods are often mixed, so if I want to find only specs for e.g. `otp` feature, I need to dig a bit.

I would like to take a stab at splitting it up, but before that I wanted to check if such a change would be accepted. Of course, we would still retain integration tests between various 2FA methods.

**Comment by jeremyevans** [comment]
I think it is fine to split off specs that use only one of the two factor features to a separate spec file for that feature.  It's probably best to keep the specs that use multiple two factor features in `two_factor_spec.rb`.
